src/parse.c

```diff
@@ -63,6 +63,45 @@ int missing_ids_found;
  * Loading and error handling
  ****************************************************/
 
+static char *
+get_error_context(yaml_parser_t parser, GError **error)
+{
+    GString *message = NULL;
+    unsigned char* line = parser.buffer.pointer;
+    unsigned char* current = line;
+    int i;
+
+    message = g_string_sized_new(200);
+
+    while (current > parser.buffer.start) {
+        current--;
+        if (*current == '\n') {
+            line = current + 2;
```

Comment:
Afaict, this line is buggy, it should just be `current + 1`.

I've tried tracing this through with sample input:
```
    end:                         v
   last:           v
  start: v
 buffer: ABC␤↹EF␤GH∅
current:     ^ (#70)
   line:     ^ (#71)
current:    ^ (#77)
   line:      ^ (#79) -- this appears buggy
current:      ^ (#85)
current:       ^ (#91)
current:        ^ (#91)
 buffer: ABC␤↹EF∅GH∅ (#88)
   line:      ^ (unchanged from above)
current:        ^ (unchanged from above)
message:…EF∅ (#94)
```